### PR TITLE
Fix/6673/blocking contact model update

### PR DESCRIPTION
**Pull request by osmaczko**
- [fix(settings/profile): remove imperative filtering from ContactPanel](https://github.com/status-im/status-desktop/pull/7044/commits/155f58d4a2d355d2c4eb2334231eb8b7a6fcc79a)
   - fixes: https://github.com/status-im/status-desktop/issues/6673
iterates: https://github.com/status-im/status-desktop/issues/6509
- [chore(settings/profile): id: contactListRoot -> id: root](https://github.com/status-im/status-desktop/pull/7044/commits/8e4d938cd72ee93a9acb15c9fd41bba1757a15c8)
- [chore(settings/profile): id: container -> id: root](https://github.com/status-im/status-desktop/pull/7044/commits/5bdf2f6627eaa3115d3879d421c7e4fb23c0a0ee)

https://user-images.githubusercontent.com/33099791/184866911-4d6d7c9a-ed07-4e93-a127-78f34b88f7d6.mp4

### CODE REVIEW

**Comment by osmaczko** [comment]
@micieslak could you please test the fix? Please keep in mind search utility may be affected as well.

**Review by jrainville — APPROVED** [review]

**Review by caybro — APPROVED** [review]
LGTM

File: ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml [hunk 1]
```diff
@@ -36,79 +37,62 @@ Item {
...
+                       pubkey.toLowerCase().includes(lowerCaseSearchString) ||
+                       compressedPubkey.toLowerCase().includes(lowerCaseSearchString)
+            }
+
+            filters: [
+                ExpressionFilter { expression: filteredModel.panelUsagePredicate(model.isVerified) },
+                ExpressionFilter {
+                    expression: {
+                        root.searchString // ensure expression is reevaluated when searchString changes
+                        filteredModel.searchPredicate(model.displayName, model.pubKey)
```
Comment (micieslak) [inline ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml:86]: shouldn't it be preceded by a `return`?

**Review by micieslak — APPROVED** [review]
Cool! Works as expected. Some minor comments.

File: ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml [hunk 2]
```diff
@@ -36,79 +37,62 @@ Item {
...
-        ScrollBar.vertical.policy: contactListRoot.scrollbarOn ? ScrollBar.AlwaysOn : ScrollBar.AsNeeded
-        model: delegateModel
-    }
+        ScrollBar.vertical.policy: root.scrollbarOn ? ScrollBar.AlwaysOn : ScrollBar.AsNeeded
+        model: SortFilterProxyModel {
+            id: filteredModel
+
+            sourceModel: root.contactsModel
+
+            function panelUsagePredicate(isVerified) {
```
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml:64]: Yeah, I know that, thanks. I find expression equally readable.

File: ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml [hunk 3]
```diff
@@ -36,79 +37,62 @@ Item {
...
+                if (panelUsage === Constants.contactsPanelUsage.verifiedMutualContacts) return isVerified
+                if (panelUsage === Constants.contactsPanelUsage.mutualContacts) return !isVerified
+                return true
+            }
+
+            function searchPredicate(name, pubkey) {
+                if (root.searchString === "") return true
```
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml]: Good point!

File: ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml (hunk 1 above)
Comment (osmaczko) [reply ui/app/AppLayouts/Profile/panels/ContactsListPanel.qml]: it should :sweat_smile: dunno how it worked before

**Pull request merged**